pkg/rules/celsig/signature.go

```diff
@@ -0,0 +1,123 @@
+package celsig
+
+import (
+	"fmt"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/vm"
+	"github.com/aquasecurity/tracee/pkg/config"
+	"github.com/aquasecurity/tracee/types/detect"
+	"github.com/aquasecurity/tracee/types/protocol"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type signature struct {
+	metadata       detect.SignatureMetadata
+	selectedEvents []detect.SignatureEventSelector
+	program        *vm.Program
+	cb             detect.SignatureHandler
+}
+
+// New constructs a Common Expression Language (CEL) signature based on the
+// specified configuration.
+//
+// Initially I wanted to use https://github.com/google/cel-go module, but I ended up
+// with https://github.com/antonmedv/expr for the POC due to /https://github.com/google/cel-go/issues/408.
+// We still need to figure out which expression language and implementation is suitable for Tracee-Rules.
+func New(config config.Signature) (detect.Signature, error) {
+	program, err := expr.Compile(config.Expression)
+	if err != nil {
+		return nil, err
+	}
+
+	var selectedEvents []detect.SignatureEventSelector
+	for _, se := range config.EventSelectors {
+		selectedEvents = append(selectedEvents, detect.SignatureEventSelector{
+			Source: se.Source,
+			Name:   se.Name,
+			Origin: se.Origin,
+		})
+	}
+
+	return &signature{
+		program:        program,
+		selectedEvents: selectedEvents,
+		metadata: detect.SignatureMetadata{
+			ID:   config.Metadata.ID,
+			Name: config.Metadata.Name,
+		},
+	}, nil
+}
+
+func (s *signature) GetMetadata() (detect.SignatureMetadata, error) {
+	return s.metadata, nil
+}
+
+func (s *signature) GetSelectedEvents() ([]detect.SignatureEventSelector, error) {
+	return s.selectedEvents, nil
+}
+
+func (s *signature) Init(cb detect.SignatureHandler) error {
+	s.cb = cb
+	return nil
+}
+
+func (s *signature) OnEvent(event protocol.Event) error {
+	env := map[string]interface{}{
+		"input":    event.Payload,
+		"argument": argument,
+		"arg":      argument,
+	}
+	out, err := expr.Run(s.program, env)
+	if err != nil {
+		return err
+	}
+	matches, ok := out.(bool)
+	if !ok {
+		return fmt.Errorf("unrecognized value: %v", out)
+	}
+	if matches {
+		s.cb(detect.Finding{
+			Event:       event,
+			SigMetadata: s.metadata,
```

Comment:
Can we add a feature to get a certain argument or value as the Data field? This could be defined in the signature